app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -0,0 +1,52 @@
+class SinglePageSubscriptionsController < ApplicationController
+  include AccountHelper
+  include GovukPersonalisation::ControllerConcern
+
+  before_action do
+    head :not_found unless govuk_account_auth_enabled?
+  end
+
+  skip_before_action :verify_authenticity_token, only: [:show]
+
+  def show
+    content_item = GdsApi.content_store.content_item(topic).to_h
+    return unless logged_in?
+
+    subscriber_list = GdsApi.email_alert_api.find_or_create_subscriber_list(
+      {
+        url: topic,
+        name: content_item["title"],
+        content_id: content_item["content_id"],
+      },
+    ).to_h.fetch("subscriber_list")
+
+    result = CreateAccountSubscriptionService.call(subscriber_list, "daily", @account_session_header)
+    account_flash_add CreateAccountSubscriptionService::SUCCESS_FLASH
+    set_account_session_header(result[:govuk_account_session])
+
+    redirect_to topic
+  rescue GdsApi::ContentStore::ItemNotFound
+    head :not_found
+  rescue GdsApi::HTTPUnauthorized
+    logout!
+    redirect_with_analytics single_page_session_path
+  end
+
+  def edit
+    redirect_with_analytics single_page_session_path
+  end
+
+private
+
+  def topic
+    @topic ||= params.fetch(:topic)
+  end
+
+  def single_page_session_path
+    redirect_path = "#{confirm_account_subscription_path}?topic=#{topic}"
+
+    GdsApi.account_api.get_sign_in_url(
+      redirect_path: redirect_path,
+    )["auth_uri"]
```

Comment:
I think you should be able to construct this without the string interpolation, like so:

```ruby
    GdsApi.account_api.get_sign_in_url(
      redirect_path: confirm_account_subscription_path(topic: topic),
    )["auth_uri"]
```

Reply:
Doing it this way gives a webmock error where it's encoding the `topic` parameter. I get: 

```
Unregistered request: GET http://account-api.dev.gov.uk/api/oauth2/sign-in?mfa=false&redirect_path=/email/subscriptions/account/confirm?topic=%252Ftest

...


registered request stubs:

       stub_request(:get, "http://account-api.dev.gov.uk/api/oauth2/sign-in?mfa=false&redirect_path=/email/subscriptions/account/confirm?topic=/test")
```
I'm not sure how the route we're passing the user on to will cope with the encoded parameter, or how to set up the stub if it's ok.

Reply:
Ok, this is probably fine then.  If something does need to be URL encoded to make it work, we'll discover that when we try things out in an actual web browser.